pkg/signup/service/signup_service.go

```diff
@@ -313,23 +313,26 @@ func (s *ServiceImpl) UpdateUserSignup(userSignup *v1alpha1.UserSignup) (*v1alph
 
 // PhoneNumberAlreadyInUse checks if the phone number has been banned. If so, return
 // an internal server error. If not, check if a signup with a different userID
-// exists. If so, return an internal server error. Otherwise, return without error.
-func (s *ServiceImpl) PhoneNumberAlreadyInUse(userID, e164PhoneNumber string) error {
-	bannedUserList, err := s.CRTClient().V1Alpha1().BannedUsers().ListByPhone(e164PhoneNumber)
+// and email address exists. If so, return an internal server error. Otherwise, return without error.
+// Either the actual phone number, or the md5 hash of the phone number may be provided here.
+func (s *ServiceImpl) PhoneNumberAlreadyInUse(userID, phoneNumberOrHash string) error {
+	bannedUserList, err := s.CRTClient().V1Alpha1().BannedUsers().ListByPhoneNumberOrHash(phoneNumberOrHash)
 	if err != nil {
 		return errs.NewInternalError(err, "failed listing banned users")
 	}
 	if len(bannedUserList.Items) > 0 {
 		return errs.NewForbiddenError("cannot re-register with phone number", "phone number already in use")
 	}
 
-	userSignupList, err := s.CRTClient().V1Alpha1().UserSignups().ListByPhone(e164PhoneNumber)
+	userSignupList, err := s.CRTClient().V1Alpha1().UserSignups().ListByPhoneNumberOrHash(phoneNumberOrHash)
 	if err != nil {
 		return errs.NewInternalError(err, "failed listing userSignups")
 	}
 	for _, signup := range userSignupList.Items {
-		if signup.Name != userID {
-			return errs.NewForbiddenError("cannot re-register with phone number", "phone number already in use")
+
+		if signup.Spec.UserID != userID && !signup.Spec.Deactivated {
```

Comment:
If you use my proposed way of using the label selector, then you can drop the check from here:
```suggestion
		if signup.Spec.UserID != userID {
```